### ISSUE TITLE
Expand auto indexing to >, >=, < and <=

### ISSUE THREAD

**Issue by arunsrinivasan** [Rdatatable/data.table#1068]

**Comment by skanskan** [comment]
Could you give more details, please?

**Comment by DavidArenburg** [comment]
+10 for that

**Comment by arunsrinivasan** [comment]
Will have to do that *after* rolling out non-equi joins.. moving to 2.0

**Comment by franknarf1** [comment]
Also, `%between%` (though is that just a special case of `inrange`, which is already listed above?).  

For example, [this SO question](http://stackoverflow.com/questions/42786896/efficiently-update-multiple-data-table-columns-based-on-a-single-condition) could be done efficiently with it, I guess. Suppose the goal is to set x to 0 outside the closed interval [-90, 90]:

    library(data.table)
    data <- data.table(x = c(10, -10, 20, -20, 100, -100))

    # this should be fairly fast if between uses auto indexing
    data[, tmp_x := numeric(.N) ][between(x, -90, 90), tmp_x := x ][, x := NULL]
    setnames(data, "tmp_x", "x")

Or even simpler if logical operations are also supported (#1453):

    data[!between(x, -90, 90), x := 0 ]

**Comment by skanskan** [comment]
maybe you mean as.numeric instead of numeric ?

**Comment by franknarf1** [comment]
@skanskan No, I'm initializing to zero with `numeric` then overwriting in the subset between -90 and 90.

**Comment by skanskan** [comment]
@franknarf1 Why not just tmp_x := 0 ?

**Comment by franknarf1** [comment]
@skanskan No reason. Sorry for the confusion.

**Comment by gregorp** [comment]
Would this allow for using `!=` as an operator in non-equi joins?

**Comment by MichaelChirico** [comment]
I think that conversely `!=` as an operator is a pre-cursor for this issue
